lib/intelligent_foods/api_client.rb

```diff
@@ -45,6 +44,25 @@ def build_request_with_body(uri:, body:)
 
     attr_reader :encoded_token, :request, :response, :uri
 
+    def handle_response(response:)
+      body = parse_response_body(response)
+      OpenStruct.new(data: body, success?: request_successful?(response.code))
+    end
+
+    def parse_response_body(response)
+      return {} if empty_response?(response.code)
+
+      JSON.parse(response.body, symbolize_names: true)
+    end
+
+    def empty_response?(response_code)
+      response_code.to_i == 204
+    end
+
+    def request_successful?(response_code)
+      response_code.to_i < 400
```

Comment:
shouldn't 3XX response codes go with the `empty_response` method?

Reply:
The API doesn't currently have any 3xx responses listed.

Should we could capture all 3xx series errors, their definitions if they should include a response body are a mixed bag.

We could also use the content-length header, however we'd need to ensure the API is providing the header accurately, and for each endpoint.